.github/workflows/main.yml

```diff
@@ -0,0 +1,34 @@
+name: Dagger/Docker Testing Suite
+
+on:
+  pull_request:
+
+concurrency:
+  # For a given workflow, if we push to the same branch, cancel all previous builds on that branch except on main.
+  group: "${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  lint:
+    name: Linting
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.13" ]
+    steps:
```

Comment:
The steps here in `workflow/main.yml` and in `templates/dagger-docker.yml` are different, but they have the same name "Dagger/Docker Testing Suite" !  I don't really know how this works but I find it odd.